chain/index/events_test.go

```diff
@@ -0,0 +1,412 @@
+	})
+	require.NoError(t, err)
+
+	ces, err = si.GetEventsForFilter(ctx, f, false)
+	require.NoError(t, err)
+	require.Equal(t, 0, len(ces))
+
+	f = &EventFilter{
+		TipsetCid: tsCid,
+	}
+	ces, err = si.GetEventsForFilter(ctx, f, false)
+	require.NoError(t, err)
+	require.Equal(t, 0, len(ces))
+
+	// search for a range that is absent
+	f = &EventFilter{
+		MinHeight: 100,
+		MaxHeight: 200,
+	}
+	ces, err = si.GetEventsForFilter(ctx, f, false)
+	require.NoError(t, err)
+	require.Equal(t, 0, len(ces))
+}
+
+func TestGetEventsForFilterWithEvents(t *testing.T) {
+	ctx := context.Background()
+	seed := time.Now().UnixNano()
+	t.Logf("seed: %d", seed)
+	rng := pseudo.New(pseudo.NewSource(seed))
+	headHeight := abi.ChainEpoch(60)
+	si, _, cs := setupWithHeadIndexed(t, headHeight, rng)
+	t.Cleanup(func() { _ = si.Close() })
+
+	ev1 := fakeEvent(
+		abi.ActorID(1),
+		[]kv{
+			{k: "type", v: []byte("approval")},
+			{k: "signer", v: []byte("addr1")},
+		},
+		[]kv{
+			{k: "amount", v: []byte("2988181")},
+		},
+	)
+
+	ev2 := fakeEvent(
+		abi.ActorID(2),
+		[]kv{
+			{k: "type", v: []byte("approval")},
+			{k: "signer", v: []byte("addr2")},
+		},
+		[]kv{
+			{k: "amount", v: []byte("2988181")},
+		},
+	)
+
+	events := []types.Event{*ev1, *ev2}
+
+	fm := fakeMessage(address.TestAddress, address.TestAddress)
+	em1 := executedMessage{
+		msg: fm,
+		evs: events,
+	}
+
+	si.SetActorToDelegatedAddresFunc(func(ctx context.Context, emitter abi.ActorID, ts *types.TipSet) (address.Address, bool) {
+		idAddr, err := address.NewIDAddress(uint64(emitter))
+		if err != nil {
+			return address.Undef, false
+		}
+
+		return idAddr, true
+	})
+
+	si.SetExecutedMessagesLoaderFunc(func(ctx context.Context, cs ChainStore, msgTs, rctTs *types.TipSet) ([]executedMessage, error) {
+		return []executedMessage{em1}, nil
+	})
+
+	// Create a fake tipset at height 1
+	fakeTipSet1 := fakeTipSet(t, rng, 1, nil)
+	fakeTipSet2 := fakeTipSet(t, rng, 2, nil)
+
+	// Set the dummy chainstore to return this tipset for height 1
+	cs.SetTipsetByHeightAndKey(1, fakeTipSet1.Key(), fakeTipSet1) // empty DB
+	cs.SetTipsetByHeightAndKey(2, fakeTipSet2.Key(), fakeTipSet2) // empty DB
+
+	cs.SetMessagesForTipset(fakeTipSet1, []types.ChainMsg{fm})
+
+	// index tipset and events
+	require.NoError(t, si.Apply(ctx, fakeTipSet1, fakeTipSet2))
+
+	// fetch it based on height -> works
+	f := &EventFilter{
+		MinHeight: 1,
+		MaxHeight: 1,
+	}
+	ces, err := si.GetEventsForFilter(ctx, f, false)
+	require.NoError(t, err)
+	require.Equal(t, 2, len(ces))
+
+	// fetch it based on cid -> works
+	tsCid1, err := fakeTipSet1.Key().Cid()
+	require.NoError(t, err)
+
+	tsCid2, err := fakeTipSet2.Key().Cid()
+	require.NoError(t, err)
+
+	f = &EventFilter{
+		TipsetCid: tsCid1,
+	}
+	ces, err = si.GetEventsForFilter(ctx, f, false)
+	require.NoError(t, err)
+	require.Equal(t, 2, len(ces))
+
+	require.Equal(t, ev1.Entries, ces[0].Entries)
+	require.Equal(t, ev2.Entries, ces[1].Entries)
```

Comment:
```suggestion
	require.Equal(t, []*CollectedEvent{
		{
			Entries:     ev1.Entries,
			EmitterAddr: must.One(address.NewIDAddress(uint64(ev1.Emitter))),
			EventIdx:    0,
			Reverted:    false,
			Height:      1,
			TipSetKey:   fakeTipSet1.Key(),
			MsgIdx:      0,
			MsgCid:      fm.Cid(),
		},
		{
			Entries:     ev2.Entries,
			EmitterAddr: must.One(address.NewIDAddress(uint64(ev2.Emitter))),
			EventIdx:    1,
			Reverted:    false,
			Height:      1,
			TipSetKey:   fakeTipSet1.Key(),
			MsgIdx:      0,
			MsgCid:      fm.Cid(),
		},
	}, ces)
```

I was messing with events.go and was surprised it didn't fail when I omitted some of the CollectedEvent properties.

But also, the above suggested change fails because `TipSetKey` is `""` for both of them because you never use `SetTipSetByCid` in `dummyChainStore`. I think `dummyChainStore#tipSetByCid` might be a remnant of a previous refactor? I'll comment over there but it would be good if you could make the above change pass.

Reply:
These tests are now green. Had to make the `SetTipsetByCid` thing work on the `dummyChainStore`.